arch/arm/core/aarch64/swap_helper.S

```diff
@@ -220,7 +209,15 @@ offload:
 	b	inv
 
 context_switch:
-	bl	z_arm64_pendsv
+	/* Check if we need to context switch */
```

Comment:
Isn't this a functional change @carlocaione ?
Sort of contradicts with the PR claim of no behavioral changes - am I missing something?

Reply:
The apparent behaviour would be the same (this prevents switching into the same context); but sure, this is more of an optimisation than a simple clean-up

Reply:
Great, let's get this captured, then, in the commit msg

Reply:
yes, it's just an optimization. I'll fix the commit msg right away.